aggregators/aggregator.go

```diff
@@ -598,6 +601,9 @@ func (a *Aggregator) processHarvest(
 	overflowLogger := nopLogger
 	if a.cfg.OverflowLogging {
 		overflowLogger = a.cfg.Logger.With(zap.Duration("aggregation_interval_ns", aggIvl))
+		if kvs := a.cfg.CombinedMetricsIDToKVs(cmk.ID); len(kvs) > 0 {
+			overflowLogger = overflowLogger.With(otelKVsToZapFields(kvs)...)
+		}
```

Comment:
```suggestion
		fields := append([]zap.Field{
			zap.Duration("aggregation_interval_ns", aggIvl),
		}, otelKVsToZapFields(kvs)...)
		overflowLogger = a.cfg.Logger.WithLazy(fields...)
```
We can use `WithLazy` for some performance improvements since `overflowLogger` will not always be used.

Reply:
Oops, missed the `kvs := a.cfg.CombinedMetricsIDToKVs(cmk.ID)` 🤦